[iOS] Issue when using BindableLayout and Entry/Editor

### Description

I'm using a StackLayout with `BindableLayout.ItemsSource` to show Entry or Editor.
However, the app throws an exception:

`System.ArgumentNullException: Value cannot be null. (Parameter 'fromPosition')`
<img width="874" alt="Screenshot 2023-06-29 at 18 44 52" src="https://github.com/dotnet/maui/assets/43689290/b7170d4e-9185-499d-9c84-b54c28c10c65">

### Steps to Reproduce

1. Create a .NET Maui project.
2. Add the following XAML code.

```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             x:Class="MauiSample.MainPage">

    <ScrollView>
        <VerticalStackLayout
            Spacing="25"
            Padding="30,0"
            VerticalOptions="Center">


            <StackLayout Margin="10"
                         VerticalOptions="Center"
                         BindableLayout.ItemsSource="{Binding CustomList}">
                <BindableLayout.ItemTemplate>
                    <DataTemplate>
                        <Grid Margin="0,0,0,20"
                              Padding="0"
                              RowSpacing="0"
                              ColumnSpacing="0">
                            <Entry Text="{Binding Name}"
                                   MaxLength="{Binding MaxLength}"
                                   Keyboard="{Binding KeyboardType}"/>

                            <!--Same issue using Editor-->
                            <!--<Editor Text="{Binding Name}"
                                   MaxLength="{Binding MaxLength}"
                                   Keyboard="{Binding KeyboardType}"/>-->
                        </Grid>
                    </DataTemplate>
                </BindableLayout.ItemTemplate>
            </StackLayout>

            <Image
                Source="dotnet_bot.png"
                SemanticProperties.Description="Cute dot net bot waving hi to you!"
                HeightRequest="200"
                HorizontalOptions="Center" />

            <Label
                Text="Hello, World!"
                SemanticProperties.HeadingLevel="Level1"
                FontSize="32"
                HorizontalOptions="Center" />

            <Label
                Text="Welcome to .NET Multi-platform App UI"
                SemanticProperties.HeadingLevel="Level2"
                SemanticProperties.Description="Welcome to dot net Multi platform App U I"
                FontSize="18"
                HorizontalOptions="Center" />

            <Button
                x:Name="CounterBtn"
                Text="Click me"
                SemanticProperties.Hint="Counts the number of times you click"
                Clicked="OnCounterClicked"
                HorizontalOptions="Center" />

        </VerticalStackLayout>
    </ScrollView>

</ContentPage>
```

3. Add the following code behind

```c#
using System.Collections.ObjectModel;

namespace MauiSample;

public class CustomItem
{
    public string Name { get; set; }
    public double MaxLength { get; set; }
    public Keyboard KeyboardType { get; set; }
}

public partial class MainPage : ContentPage
{
	int count = 0;
	public ObservableCollection<CustomItem> CustomList { get; set; }

	public MainPage()
	{
		InitializeComponent();
		CustomList = new ObservableCollection<CustomItem>();
		AddCustomItems();
        BindingContext = this;
    }

	void AddCustomItems()
	{
		CustomList.Clear();

        for (int i = 0; i < 10; i++)
        {
            CustomList.Add(new CustomItem
            {
                MaxLength = 10,
                KeyboardType = Keyboard.Email,
                Name = "my value: " + i
            });
        }


  //      CustomList.Add(new CustomItem { Name = "Test1", MaxLength = 20, KeyboardType = Keyboard.Email });
		//CustomList.Add(new CustomItem { Name = "Test2", MaxLength = 30, KeyboardType = Keyboard.Plain });
		//CustomList.Add(new CustomItem { Name = "Test3", MaxLength = 10, KeyboardType = Keyboard.Numeric });
		//CustomList.Add(new CustomItem { Name = "Test4", MaxLength = 23, KeyboardType = Keyboard.Default });
    }

	private void OnCounterClicked(object sender, EventArgs e)
	{
		count++;

		if (count == 1)
			CounterBtn.Text = $"Clicked {count} time";
		else
			CounterBtn.Text = $"Clicked {count} times";

		SemanticScreenReader.Announce(CounterBtn.Text);
	}
}
```

4. Run the project and the app will crash.

### Link to public reproduction project repository

https://github.com/juniorsaraviao/MauiSample

### Version with bug

7.0.86

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15.2 and up

### Did you find any workaround?

No

### Relevant log output

```shell
Unhandled Exception:
System.ArgumentNullException: Value cannot be null. (Parameter 'fromPosition')
   at ObjCRuntime.ThrowHelper.ThrowArgumentNullException(String argumentName)
   at ObjCRuntime.NativeObjectExtensions.GetNonNullHandle(INativeObject self, String argumentName)
   at UIKit.UITextField.GetTextRange(UITextPosition fromPosition, UITextPosition toPosition)
   at Microsoft.Maui.Platform.TextInputExtensions.SetTextRange(IUITextInput platformView, Int32 start, Int32 selectedTextLength)
   at Microsoft.Maui.Controls.Platform.TextExtensions.UpdateText(UITextField textField, InputView inputView)
   at Microsoft.Maui.Controls.Entry.MapText(IEntryHandler handler, Entry entry)
   at Microsoft.Maui.Controls.Entry.MapText(EntryHandler handler, Entry entry)
   at Microsoft.Maui.PropertyMapper`2.<>c__DisplayClass5_0[[Microsoft.Maui.Controls.Entry, Microsoft.Maui.Controls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Handlers.EntryHandler, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<Add>b__0(IElementHandler h, IElement v)
   at Microsoft.Maui.PropertyMapper.UpdatePropertyCore(String key, IElementHandler viewHandler, IElement virtualView)
   at Microsoft.Maui.PropertyMapper.UpdateProperties(IElementHandler viewHandler, IElement virtualView)
   at Microsoft.Maui.Handlers.ElementHandler.SetVirtualView(IElement view)
   at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IEntry, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.MauiTextField, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IView view)
   at Microsoft.Maui.Handlers.EntryHandler.SetVirtualView(IView view)
   at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IEntry, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.MauiTextField, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
   at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
   at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
   at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
   at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
   at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
   at Microsoft.Maui.Handlers.LayoutHandler.SetVirtualView(IView view)
   at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.ILayout, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.LayoutView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
   at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
   at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
   at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
   at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
   at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
   at Microsoft.Maui.Handlers.LayoutHandler.SetVirtualView(IView view)
   at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.ILayout, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.LayoutView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
   at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
   at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
   at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
   at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
   at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
   at Microsoft.Maui.Handlers.LayoutHandler.SetVirtualView(IView view)
   at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.ILayout, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.LayoutView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
   at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
   at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
   at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
   at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
   at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
   at Microsoft.Maui.Handlers.ScrollViewHandler.UpdateContentView(IScrollView scrollView, IScrollViewHandler handler)
   at Microsoft.Maui.Handlers.ScrollViewHandler.MapContent(IScrollViewHandler handler, IScrollView scrollView)
   at Microsoft.Maui.PropertyMapper`2.<>c__DisplayClass5_0[[Microsoft.Maui.IScrollView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Handlers.IScrollViewHandler, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<Add>b__0(IElementHandler h, IElement v)
   at Microsoft.Maui.PropertyMapper.UpdatePropertyCore(String key, IElementHandler viewHandler, IElement virtualView)
   at Microsoft.Maui.PropertyMapper.UpdateProperties(IElementHandler viewHandler, IElement virtualView)
   at Microsoft.Maui.Handlers.ElementHandler.SetVirtualView(IElement view)
   at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IScrollView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[UIKit.UIScrollView, Microsoft.iOS, Version=16.4.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065]].SetVirtualView(IView view)
   at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IScrollView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[UIKit.UIScrollView, Microsoft.iOS, Version=16.4.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065]].SetVirtualView(IElement view)
   at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
   at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
   at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
   at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
   at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context)
   at Microsoft.Maui.Handlers.ContentViewHandler.UpdateContent(IContentViewHandler handler)
   at Microsoft.Maui.Handlers.ContentViewHandler.MapContent(IContentViewHandler handler, IContentView page)
   at Microsoft.Maui.PropertyMapper`2.<>c__DisplayClass5_0[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Handlers.IContentViewHandler, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<Add>b__0(IElementHandler h, IElement v)
   at Microsoft.Maui.PropertyMapper.UpdatePropertyCore(String key, IElementHandler viewHandler, IElement virtualView)
   at Microsoft.Maui.PropertyMapper.UpdateProperties(IElementHandler viewHandler, IElement virtualView)
   at Microsoft.Maui.Handlers.ElementHandler.SetVirtualView(IElement view)
   at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.ContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IView view)
   at Microsoft.Maui.Handlers.ContentViewHandler.SetVirtualView(IView view)
   at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.ContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view)
   at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler)
   at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value)
   at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value)
   at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context)
   at Microsoft.Maui.Platform.ViewExtensions.ToHandler(IView view, IMauiContext context)
   at Microsoft.Maui.Controls.Platform.Compatibility.ShellSectionRootRenderer.SetPageRenderer(Page page, ShellContent shellContent)
   at Microsoft.Maui.Controls.Platform.Compatibility.ShellSectionRootRenderer.LoadRenderers()
   at Microsoft.Maui.Controls.Platform.Compatibility.ShellSectionRootRenderer.ViewDidLoad()
   at UIKit.UIApplication.UIApplicationMain(Int32 argc, String[] argv, IntPtr principalClassName, IntPtr delegateClassName)
   at UIKit.UIApplication.Main(String[] args, Type principalClass, Type delegateClass)
   at MauiSample.Program.Main(String[] args) in /Users/juniorsaraviao/Projects/MauiSample/Platforms/iOS/Program.cs:line 13
```

Additional Information:
This issue also occurs on Android. Below is the exception information.

[HostConnection] HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit ANDROID_EMU_sync_buffer_data ANDROID_EMU_read_color_buffer_dma GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_gles_max_version_2 
**Java.Lang.IndexOutOfBoundsException:** 'setSpan (11 ... 11) ends beyond length 10'

**Java.Lang.IndexOutOfBoundsException:** 'setSpan (11 ... 11) ends beyond length 10'

[mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: Java.Lang.IndexOutOfBoundsException: setSpan (11 ... 11) ends beyond length 10
[mono-rt]    at Java.Interop.JniEnvironment.InstanceMethods.CallNonvirtualVoidMethod(JniObjectReference instance, JniObjectReference type, JniMethodInfo method, JniArgumentValue* args) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/obj/Release/net7.0/JniEnvironment.g.cs:line 12324
[mono-rt]    at Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeVirtualVoidMethod(String encodedMember, IJavaPeerable self, JniArgumentValue* parameters) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.JniInstanceMethods_Invoke.cs:line 75
[mono-rt]    at Android.Widget.EditText.SetSelection(Int32 index) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net7.0/android-33/mcw/Android.Widget.EditText.cs:line 208
[mono-rt]    at Microsoft.Maui.Controls.Platform.EditTextExtensions.UpdateText(EditText editText, InputView inputView) in C:\dotnet-maui-work\src\Controls\src\Core\Platform\Android\Extensions\EditTextExtensions.cs:line 46
[mono-rt]    at Microsoft.Maui.Controls.Editor.MapText(IEditorHandler handler, Editor editor) in C:\dotnet-maui-work\src\Controls\src\Core\Editor\Editor.Android.cs:line 21
[mono-rt]    at Microsoft.Maui.Controls.Editor.MapText(EditorHandler handler, Editor editor) in C:\dotnet-maui-work\src\Controls\src\Core\Editor\Editor.Android.cs:line 16
[mono-rt]    at Microsoft.Maui.PropertyMapper`2.<>c__DisplayClass5_0[[Microsoft.Maui.Controls.Editor, Microsoft.Maui.Controls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Handlers.EditorHandler, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<Add>b__0(IElementHandler h, IElement v) in C:\dotnet-maui-work\src\Core\src\PropertyMapper.cs:line 172
[mono-rt]    at Microsoft.Maui.PropertyMapper.UpdatePropertyCore(String key, IElementHandler viewHandler, IElement virtualView) in C:\dotnet-maui-work\src\Core\src\PropertyMapper.cs:line 47
[mono-rt]    at Microsoft.Maui.PropertyMapper.UpdateProperties(IElementHandler viewHandler, IElement virtualView) in C:\dotnet-maui-work\src\Core\src\PropertyMapper.cs:line 82
[mono-rt]    at Microsoft.Maui.Handlers.ElementHandler.SetVirtualView(IElement view) in C:\dotnet-maui-work\src\Core\src\Handlers\Element\ElementHandler.cs:line 79
[mono-rt]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IEditor, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[AndroidX.AppCompat.Widget.AppCompatEditText, Xamarin.AndroidX.AppCompat, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IView view) in C:\dotnet-maui-work\src\Core\src\Handlers\View\ViewHandlerOfT.cs:line 53
[mono-rt]    at Microsoft.Maui.Handlers.EditorHandler.SetVirtualView(IView view) in C:\dotnet-maui-work\src\Core\src\Handlers\Editor\EditorHandler.Android.cs:line 34
[mono-rt]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IEditor, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[AndroidX.AppCompat.Widget.AppCompatEditText, Xamarin.AndroidX.AppCompat, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view) in C:\dotnet-maui-work\src\Core\src\Handlers\View\ViewHandlerOfT.cs:line 56
[mono-rt]    at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 831
[mono-rt]    at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 781
[mono-rt]    at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\VisualElement\VisualElement.cs:line 1660
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 96
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 127
[mono-rt]    at Microsoft.Maui.Handlers.LayoutHandler.SetVirtualView(IView view) in C:\dotnet-maui-work\src\Core\src\Handlers\Layout\LayoutHandler.Android.cs:line 43
[mono-rt]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.ILayout, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.LayoutViewGroup, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view) in C:\dotnet-maui-work\src\Core\src\Handlers\View\ViewHandlerOfT.cs:line 56
[mono-rt]    at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 831
[mono-rt]    at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 781
[mono-rt]    at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\VisualElement\VisualElement.cs:line 1660
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 96
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 127
[mono-rt]    at Microsoft.Maui.Handlers.LayoutHandler.SetVirtualView(IView view) in C:\dotnet-maui-work\src\Core\src\Handlers\Layout\LayoutHandler.Android.cs:line 43
[mono-rt]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.ILayout, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.LayoutViewGroup, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view) in C:\dotnet-maui-work\src\Core\src\Handlers\View\ViewHandlerOfT.cs:line 56
[mono-rt]    at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 831
[mono-rt]    at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 781
[mono-rt]    at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\VisualElement\VisualElement.cs:line 1660
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 96
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 127
[mono-rt]    at Microsoft.Maui.Handlers.LayoutHandler.SetVirtualView(IView view) in C:\dotnet-maui-work\src\Core\src\Handlers\Layout\LayoutHandler.Android.cs:line 43
[mono-rt]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.ILayout, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.LayoutViewGroup, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view) in C:\dotnet-maui-work\src\Core\src\Handlers\View\ViewHandlerOfT.cs:line 56
[mono-rt]    at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 831
[mono-rt]    at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 781
[mono-rt]    at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\VisualElement\VisualElement.cs:line 1660
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 96
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 127
[mono-rt]    at Microsoft.Maui.Handlers.ScrollViewHandler.UpdateInsetView(IScrollView scrollView, IScrollViewHandler handler) in C:\dotnet-maui-work\src\Core\src\Handlers\ScrollView\ScrollViewHandler.Android.cs:line 175
[mono-rt]    at Microsoft.Maui.Handlers.ScrollViewHandler.MapContent(IScrollViewHandler handler, IScrollView scrollView) in C:\dotnet-maui-work\src\Core\src\Handlers\ScrollView\ScrollViewHandler.Android.cs:line 109
[mono-rt]    at Microsoft.Maui.PropertyMapper`2.<>c__DisplayClass5_0[[Microsoft.Maui.IScrollView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Handlers.IScrollViewHandler, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<Add>b__0(IElementHandler h, IElement v) in C:\dotnet-maui-work\src\Core\src\PropertyMapper.cs:line 172
[mono-rt]    at Microsoft.Maui.PropertyMapper.UpdatePropertyCore(String key, IElementHandler viewHandler, IElement virtualView) in C:\dotnet-maui-work\src\Core\src\PropertyMapper.cs:line 47
[mono-rt]    at Microsoft.Maui.PropertyMapper.UpdateProperties(IElementHandler viewHandler, IElement virtualView) in C:\dotnet-maui-work\src\Core\src\PropertyMapper.cs:line 82
[mono-rt]    at Microsoft.Maui.Handlers.ElementHandler.SetVirtualView(IElement view) in C:\dotnet-maui-work\src\Core\src\Handlers\Element\ElementHandler.cs:line 79
[mono-rt]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IScrollView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.MauiScrollView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IView view) in C:\dotnet-maui-work\src\Core\src\Handlers\View\ViewHandlerOfT.cs:line 53
[mono-rt]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IScrollView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.MauiScrollView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view) in C:\dotnet-maui-work\src\Core\src\Handlers\View\ViewHandlerOfT.cs:line 56
[mono-rt]    at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 831
[mono-rt]    at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 781
[mono-rt]    at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\VisualElement\VisualElement.cs:line 1660
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 96
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 127
[mono-rt]    at Microsoft.Maui.Handlers.ContentViewHandler.UpdateContent(IContentViewHandler handler) in C:\dotnet-maui-work\src\Core\src\Handlers\ContentView\ContentViewHandler.Android.cs:line 44
[mono-rt]    at Microsoft.Maui.Handlers.ContentViewHandler.MapContent(IContentViewHandler handler, IContentView page) in C:\dotnet-maui-work\src\Core\src\Handlers\ContentView\ContentViewHandler.Android.cs:line 49
[mono-rt]    at Microsoft.Maui.PropertyMapper`2.<>c__DisplayClass5_0[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Handlers.IContentViewHandler, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<Add>b__0(IElementHandler h, IElement v) in C:\dotnet-maui-work\src\Core\src\PropertyMapper.cs:line 172
[mono-rt]    at Microsoft.Maui.PropertyMapper.UpdatePropertyCore(String key, IElementHandler viewHandler, IElement virtualView) in C:\dotnet-maui-work\src\Core\src\PropertyMapper.cs:line 47
[mono-rt]    at Microsoft.Maui.PropertyMapper.UpdateProperties(IElementHandler viewHandler, IElement virtualView) in C:\dotnet-maui-work\src\Core\src\PropertyMapper.cs:line 82
[mono-rt]    at Microsoft.Maui.Handlers.ElementHandler.SetVirtualView(IElement view) in C:\dotnet-maui-work\src\Core\src\Handlers\Element\ElementHandler.cs:line 79
[mono-rt]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.ContentViewGroup, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IView view) in C:\dotnet-maui-work\src\Core\src\Handlers\View\ViewHandlerOfT.cs:line 53
[mono-rt]    at Microsoft.Maui.Handlers.ContentViewHandler.SetVirtualView(IView view) in C:\dotnet-maui-work\src\Core\src\Handlers\ContentView\ContentViewHandler.Android.cs:line 27
[mono-rt]    at Microsoft.Maui.Handlers.ViewHandler`2[[Microsoft.Maui.IContentView, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null],[Microsoft.Maui.Platform.ContentViewGroup, Microsoft.Maui, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetVirtualView(IElement view) in C:\dotnet-maui-work\src\Core\src\Handlers\View\ViewHandlerOfT.cs:line 56
[mono-rt]    at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 831
[mono-rt]    at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\Element\Element.cs:line 781
[mono-rt]    at Microsoft.Maui.Controls.VisualElement.Microsoft.Maui.IElement.set_Handler(IElementHandler value) in C:\dotnet-maui-work\src\Controls\src\Core\VisualElement\VisualElement.cs:line 1660
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToHandler(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 96
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.ToPlatform(IElement view, IMauiContext context) in C:\dotnet-maui-work\src\Core\src\Platform\ElementExtensions.cs:line 127
[mono-rt]    at Microsoft.Maui.Platform.MauiContextExtensions.ToPlatform(IView view, IMauiContext fragmentMauiContext, Context context, LayoutInflater layoutInflater, FragmentManager childFragmentManager) in C:\dotnet-maui-work\src\Core\src\Platform\Android\MauiContextExtensions.cs:line 96
[mono-rt]    at Microsoft.Maui.Controls.Platform.Compatibility.ShellFragmentContainer.OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) in C:\dotnet-maui-work\src\Controls\src\Core\Compatibility\Handlers\Shell\Android\ShellFragmentContainer.cs:line 26
[mono-rt]    at AndroidX.Fragment.App.Fragment.n_OnCreateView_Landroid_view_LayoutInflater_Landroid_view_ViewGroup_Landroid_os_Bundle_(IntPtr jnienv, IntPtr native__this, IntPtr native_inflater, IntPtr native_container, IntPtr native_savedInstanceState) in C:\a\_work\1\s\generated\androidx.fragment.fragment\obj\Release
et6.0-android\generated\src\AndroidX.Fragment.App.Fragment.cs:line 2031
[mono-rt]    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPLLL_L(_JniMarshal_PPLLL_L callback, IntPtr jnienv, IntPtr klazz, IntPtr p0, IntPtr p1, IntPtr p2) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 352
[mono-rt]   --- End of managed Java.Lang.IndexOutOfBoundsException stack trace ---
[mono-rt] java.lang.IndexOutOfBoundsException: setSpan (11 ... 11) ends beyond length 10
[mono-rt] 	at android.text.SpannableStringBuilder.checkRange(SpannableStringBuilder.java:1325)
[mono-rt] 	at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:684)
[mono-rt] 	at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:676)
[mono-rt] 	at androidx.emoji2.text.SpannableBuilder.setSpan(SpannableBuilder.java:140)
[mono-rt] 	at android.text.Selection.setSelection(Selection.java:94)
[mono-rt] 	at android.text.Selection.setSelection(Selection.java:78)
[mono-rt] 	at android.text.Selection.setSelection(Selection.java:153)
[mono-rt] 	at android.widget.EditText.setSelection(EditText.java:135)
[mono-rt] 	at crc640ec207abc449b2ca.ShellFragmentContainer.n_onCreateView(Native Method)
[mono-rt] 	at crc640ec207abc449b2ca.ShellFragmentContainer.onCreateView(ShellFragmentContainer.java:41)
[mono-rt] 	at androidx.fragment.app.Fragment.performCreateView(Fragment.java:3113)
[mono-rt] 	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:524)
[mono-rt] 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
[mono-rt] 	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1901)
[mono-rt] 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1819)
[mono-rt] 	at androidx.fragment.app.FragmentManager.execSingleAction(FragmentManager.java:1731)
[mono-rt] 	at androidx.fragment.app.BackStackRecord.commitNow(BackStackRecord.java:317)
[mono-rt] 	at androidx.viewpager2.adapter.FragmentStateAdapter.placeFragmentInViewHolder(FragmentStateAdapter.java:341)
[mono-rt] 	at androidx.viewpager2.adapter.FragmentStateAdapter.onViewAttachedToWindow(FragmentStateAdapter.java:276)
[mono-rt] 	at androidx.viewpager2.adapter.FragmentStateAdapter.onViewAttachedToWindow(FragmentStateAdapter.java:67)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView.dispatchChildAttached(RecyclerView.java:8291)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView$5.addView(RecyclerView.java:924)
[mono-rt] 	at androidx.recyclerview.widget.ChildHelper.addView(ChildHelper.java:107)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView$LayoutManager.addViewInt(RecyclerView.java:9344)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView$LayoutManager.addView(RecyclerView.java:9302)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView$LayoutManager.addView(RecyclerView.java:9289)
[mono-rt] 	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1676)
[mono-rt] 	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1622)
[mono-rt] 	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:687)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4604)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4307)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4873)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at androidx.viewpager2.widget.ViewPager2.onLayout(ViewPager2.java:527)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at com.google.android.material.appbar.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:149)
[mono-rt] 	at com.google.android.material.appbar.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:43)
[mono-rt] 	at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onLayoutChild(AppBarLayout.java:2376)
[mono-rt] 	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:953)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
[mono-rt] 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
[mono-rt] 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at androidx.drawerlayout.widget.DrawerLayout.onLayout(DrawerLayout.java:1273)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
[mono-rt] 	at android.widget.LinearLayout.layoutHorizontal(LinearLayout.java:1818)
[mono-rt] 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1584)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
[mono-rt] 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
[mono-rt] 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at com.android.internal.policy.DecorView.onLayout(DecorView.java:784)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3470)
[mono-rt] 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2938)
[mono-rt] 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1952)
[mono-rt] 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8171)
[mono-rt] 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:972)
[mono-rt] 	at android.view.Choreographer.doCallbacks(Choreographer.java:796)
[mono-rt] 	at android.view.Choreographer.doFrame(Choreographer.java:731)
[mono-rt] 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:957)
[mono-rt] 	at android.os.Handler.handleCallback(Handler.java:938)
[mono-rt] 	at android.os.Handler.dispatchMessage(Handler.java:99)
[mono-rt] 	at android.os.Looper.loop(Looper.java:223)
[mono-rt] 	at android.app.ActivityThread.main(ActivityThread.java:7656)
[mono-rt] 	at java.lang.reflect.Method.invoke(Native Method)
[mono-rt] 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
[mono-rt] 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
[mono-rt] 
[mono-rt]   --- End of managed Java.Lang.IndexOutOfBoundsException stack trace ---
[mono-rt] java.lang.IndexOutOfBoundsException: setSpan (11 ... 11) ends beyond length 10
[mono-rt] 	at android.text.SpannableStringBuilder.checkRange(SpannableStringBuilder.java:1325)
[mono-rt] 	at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:684)
[mono-rt] 	at android.text.SpannableStringBuilder.setSpan(SpannableStringBuilder.java:676)
[mono-rt] 	at androidx.emoji2.text.SpannableBuilder.setSpan(SpannableBuilder.java:140)
[mono-rt] 	at android.text.Selection.setSelection(Selection.java:94)
[mono-rt] 	at android.text.Selection.setSelection(Selection.java:78)
[mono-rt] 	at android.text.Selection.setSelection(Selection.java:153)
[mono-rt] 	at android.widget.EditText.setSelection(EditText.java:135)
[mono-rt] 	at crc640ec207abc449b2ca.ShellFragmentContainer.n_onCreateView(Native Method)
[mono-rt] 	at crc640ec207abc449b2ca.ShellFragmentContainer.onCreateView(ShellFragmentContainer.java:41)
[mono-rt] 	at androidx.fragment.app.Fragment.performCreateView(Fragment.java:3113)
[mono-rt] 	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:524)
[mono-rt] 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
[mono-rt] 	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1901)
[mono-rt] 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1819)
[mono-rt] 	at androidx.fragment.app.FragmentManager.execSingleAction(FragmentManager.java:1731)
[mono-rt] 	at androidx.fragment.app.BackStackRecord.commitNow(BackStackRecord.java:317)
[mono-rt] 	at androidx.viewpager2.adapter.FragmentStateAdapter.placeFragmentInViewHolder(FragmentStateAdapter.java:341)
[mono-rt] 	at androidx.viewpager2.adapter.FragmentStateAdapter.onViewAttachedToWindow(FragmentStateAdapter.java:276)
[mono-rt] 	at androidx.viewpager2.adapter.FragmentStateAdapter.onViewAttachedToWindow(FragmentStateAdapter.java:67)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView.dispatchChildAttached(RecyclerView.java:8291)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView$5.addView(RecyclerView.java:924)
[mono-rt] 	at androidx.recyclerview.widget.ChildHelper.addView(ChildHelper.java:107)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView$LayoutManager.addViewInt(RecyclerView.java:9344)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView$LayoutManager.addView(RecyclerView.java:9302)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView$LayoutManager.addView(RecyclerView.java:9289)
[mono-rt] 	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1676)
[mono-rt] 	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1622)
[mono-rt] 	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:687)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4604)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:4307)
[mono-rt] 	at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4873)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at androidx.viewpager2.widget.ViewPager2.onLayout(ViewPager2.java:527)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at com.google.android.material.appbar.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:149)
[mono-rt] 	at com.google.android.material.appbar.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:43)
[mono-rt] 	at com.google.android.material.appbar.AppBarLayout$ScrollingViewBehavior.onLayoutChild(AppBarLayout.java:2376)
[mono-rt] 	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:953)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
[mono-rt] 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
[mono-rt] 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at androidx.drawerlayout.widget.DrawerLayout.onLayout(DrawerLayout.java:1273)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
[mono-rt] 	at android.widget.LinearLayout.layoutHorizontal(LinearLayout.java:1818)
[mono-rt] 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1584)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
[mono-rt] 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
[mono-rt] 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
[mono-rt] 	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
[mono-rt] 	at com.android.internal.policy.DecorView.onLayout(DecorView.java:784)
[mono-rt] 	at android.view.View.layout(View.java:22844)
[mono-rt] 	at android.view.ViewGroup.layout(ViewGroup.java:6389)
[mono-rt] 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:3470)
[mono-rt] 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2938)
[mono-rt] 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1952)
[mono-rt] 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:8171)
[mono-rt] 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:972)
[mono-rt] 	at android.view.Choreographer.doCallbacks(Choreographer.java:796)
[mono-rt] 	at android.view.Choreographer.doFrame(Choreographer.java:731)
[mono-rt] 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:957)
[mono-rt] 	at android.os.Handler.handleCallback(Handler.java:938)
[mono-rt] 	at android.os.Handler.dispatchMessage(Handler.java:99)
[mono-rt] 	at android.os.Looper.loop(Looper.java:223)
[mono-rt] 	at android.app.ActivityThread.main(ActivityThread.java:7656)
[mono-rt] 	at java.lang.reflect.Method.invoke(Native Method)
[mono-rt] 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
[mono-rt] 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
[mono-rt] 
[HostConnection] HostConnection::get() New Host Connection established 0xe8f7b420, tid 17742
[HostConnection] HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit ANDROID_EMU_sync_buffer_data ANDROID_EMU_read_color_buffer_dma GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_gles_max_version_2

--- a/src/Controls/samples/Controls.Sample.UITests/Issues/Issue20439.xaml
+++ b/src/Controls/samples/Controls.Sample.UITests/Issues/Issue20439.xaml
@@ -1,0 +1,25 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<Shell xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="Maui.Controls.Sample.Issues.Issue20439"
+             xmlns:ns="clr-namespace:Maui.Controls.Sample.Issues">
+    <ShellContent
+            Title="Home">
+            <ContentPage>
+                <StackLayout VerticalOptions="Center">
+                    <Entry AutomationId="entry" Text="Tap here"/>
+                    <Button AutomationId="button" Text="Open the input page" Clicked="Button_Clicked"/>
+                </StackLayout>
+            </ContentPage>
+        </ShellContent>
+
+        <ShellContent
+            Title="Text input page">
+            <ContentPage>
+                   <Editor VerticalOptions="Center"
+                           AutomationId="editor"
+                           HeightRequest="100"
+                           Text="Potential auto correcting words: -- :-) ... omw \n" />
+            </ContentPage>
+        </ShellContent>
+</Shell>

--- a/src/Controls/samples/Controls.Sample.UITests/Issues/Issue20439.xaml.cs
+++ b/src/Controls/samples/Controls.Sample.UITests/Issues/Issue20439.xaml.cs
@@ -1,0 +1,20 @@
+﻿using Microsoft.Maui.Controls;
+using Microsoft.Maui.Controls.Xaml;
+
+namespace Maui.Controls.Sample.Issues
+{
+	[XamlCompilation(XamlCompilationOptions.Compile)]
+	[Issue(IssueTracker.Github, 20439, "[iOS] Double dash in Entry or Editor crashes the app", PlatformAffected.iOS)]
+	public partial class Issue20439 : Shell
+	{
+		public Issue20439()
+		{
+			InitializeComponent();
+		}
+
+		void Button_Clicked(object sender, System.EventArgs e)
+		{
+			CurrentItem = Items[1];
+		}
+	}
+}

--- a/src/Controls/src/Core/Platform/iOS/Extensions/TextExtensions.cs
+++ b/src/Controls/src/Core/Platform/iOS/Extensions/TextExtensions.cs
@@ -41,6 +41,12 @@ namespace Microsoft.Maui.Controls.Platform
 			// position if needed when the text was modified by a Converter.
 			var textRange = textInput.GetTextRange(textInput.BeginningOfDocument, textInput.EndOfDocument);
 			var oldText = textInput.TextInRange(textRange) ?? string.Empty;
+
+			// We need this variable because in some cases because of the iOS's
+			// auto correction eg. eg '--' => '—' the actual text in the input might have
+			// a different length that the one that has been set in the control.
+			var newTextLength = textInput.TextInRange(textRange)?.Length ?? 0;
+
 			var newText = TextTransformUtilites.GetTransformedText(
 				inputView?.Text,
 				textInput.GetSecureTextEntry() ? TextTransform.Default : inputView.TextTransform
@@ -50,8 +56,8 @@ namespace Microsoft.Maui.Controls.Platform
 			{
 				// Re-calculate the cursor offset position if the text was modified by a Converter.
 				// but if the text is being set by code, let's just move the cursor to the end.
-				var cursorOffset = newText.Length - oldText.Length;
-				var cursorPosition = isEditing ? textInput.GetCursorPosition(cursorOffset) : newText.Length;
+				var cursorOffset = newTextLength - oldText.Length;
+				var cursorPosition = isEditing ? textInput.GetCursorPosition(cursorOffset) : newTextLength;
 
 				textInput.ReplaceText(textRange, newText);
 

--- a/src/Controls/src/Core/Platform/iOS/Extensions/TextExtensions.cs
+++ b/src/Controls/src/Core/Platform/iOS/Extensions/TextExtensions.cs
@@ -45,12 +45,15 @@ namespace Microsoft.Maui.Controls.Platform
 			// We need this variable because in some cases because of the iOS's
 			// auto correction eg. eg '--' => '—' the actual text in the input might have
 			// a different length that the one that has been set in the control.
-			var newTextLength = textInput.TextInRange(textRange)?.Length ?? 0;
+			var newTextLength = !string.IsNullOrWhiteSpace(inputView.Text) ? textInput.TextInRange(textRange)?.Length ?? 0 : 0;
 
 			var newText = TextTransformUtilites.GetTransformedText(
 				inputView?.Text,
 				textInput.GetSecureTextEntry() ? TextTransform.Default : inputView.TextTransform
 				);
+
+			if (!string.IsNullOrWhiteSpace(oldText))
+				newTextLength = newText.Length;
 
 			if (oldText != newText)
 			{

--- a/src/Controls/tests/UITests/Tests/Issues/Issue20439.cs
+++ b/src/Controls/tests/UITests/Tests/Issues/Issue20439.cs
@@ -1,0 +1,28 @@
+﻿using NUnit.Framework;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.AppiumTests.Issues
+{
+	public class Issue20439 : _IssuesUITest
+	{
+		public override string Issue => "[iOS] Double dash in Entry or Editor crashes the app";
+
+		public Issue20439(TestDevice device) : base(device)
+		{
+		}
+
+		[Test]
+		public void ErrorShouldNotBeThrown()
+		{
+			this.IgnoreIfPlatforms(new TestDevice[] { TestDevice.Android, TestDevice.Mac, TestDevice.Windows });
+
+			_ = App.WaitForElement("entry");
+			App.Click("entry");
+			App.Click("button");
+
+			// The test passes if no crash is observed
+			App.FindElement("editor");
+		}
+	}
+}

